svm/src/loader.rs

```diff
@@ -23,9 +29,235 @@ pub trait Loader {
             .and_then(|account| owners.iter().position(|entry| account.owner() == entry))
     }
 
+    /// Check whether the payer_account is capable of paying the fee. The
+    /// side effect is to subtract the fee amount from the payer_account
+    /// balance of lamports. If the payer_acount is not able to pay the
+    /// fee, the error_metrics is incremented, and a specific error is
+    /// returned.
+    ///
+    /// This function has a default implementation, but projects can override
+    /// it if they want to provide a custom implementation.
+    fn validate_fee_payer(
```

Comment:
It's a convenient place to put this function. But, it makes trait a bit ambiguous. If want to add it here, we should find a different name for the trait. It's no longer account specific.